src/main/scala/com/cloudant/clouseau/IndexCleanupService.scala

```diff
@@ -65,17 +65,18 @@ class IndexCleanupService(ctx: ServiceContext[ConfigurationArgs]) extends Servic
         case 'ok =>
           'ok
         case ('error, 'not_found) =>
-          recursivelyDelete(fileOrDir)
+          recursivelyDelete(fileOrDir, false)
           fileOrDir.delete
       }
     }
   }
 
-  private def recursivelyDelete(fileOrDir: File) {
+  private def recursivelyDelete(fileOrDir: File, deleteDir: Boolean) {
     if (fileOrDir.isDirectory) {
       for (file <- fileOrDir.listFiles)
-        recursivelyDelete(file)
-      fileOrDir.delete
+        recursivelyDelete(file, deleteDir)
```

Comment:
I think this should actually always specify `deleteDir = true`, regardless of the `recursivelyDelete` caller-passed value. (At least in this case.)

The key issue arises from deleting the top-level directory (i.e., `${SEARCH_ROOT}/shards/${RANGE}/${DB_NAME_AND_SUFFIX}`) if it temporarily becomes empty during a user changing a ddoc (e.g., when clouseau is cleaning up an orphaned index signature concurrently with trying to create a directory for a new index signature). However, we do typically want to clean-up the empty `${DDOC_INDEX_SIG}` directories because deletion only takes place if a signature has changed.

Passing `deleteDir = false` ensures that any user changing their ddocs burns through inodes by leaving the subdirectories associated with the orphaned signatures on disk.

However, this whole thing is subtle, and I'm happy to hear it argued differently. :)

Reply:
Thinking about this also made me raise #55 because I think this may be a damned if we do/damned if we don't situation: regardless of the value of `deleteDir` when recursing, I believe it's possible for `CleanupDbMsg` to actually delete newly created Lucene files.

Reply:
> I guess it depends on the usecase.

Not really, it's partly about correctness and partly about maintaining stable behaviour for large-scale deployments (as with the one I'm considering).

I understand that your current use case is different, but effectively the change introduces a regression for users who _do_ update their index definitions.

Reply:
You're right, sorry. (I lost track of the fact that the subdirectories weren't deleted before.)

I'm okay with this!